controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -321,27 +321,32 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
 
     private CompletableFuture<Void> truncate(String scope, String stream, RetentionPolicy policy, OperationContext context,
                                              RetentionSet retentionSet, StreamCutRecord newRecord, long recordingTime, long requestId) {
-        return findTruncationRecord(policy, retentionSet, newRecord, recordingTime)
-                .map(record -> streamMetadataStore.getStreamCutRecord(scope, stream, record, context, executor)
-                        .thenCompose(streamCutRecord -> startTruncation(scope, stream, streamCutRecord.getStreamCut(), context, requestId))
-                        .thenCompose(started -> {
-                            if (started) {
-                                return streamMetadataStore.deleteStreamCutBefore(scope, stream, record, context, executor);
+        Optional<StreamCutReferenceRecord> truncationRecord = findTruncationRecord(policy, retentionSet, newRecord, recordingTime);
+        if (!truncationRecord.isPresent()) {
+            log.info("Could not find truncation record for stream {}/{}", scope, stream);
```

Comment:
nit: Could not find truncation record seems negative. 
I would reword it to something along the lines of - `No suitable truncation record was found that satisfied retention policy`.

Reply:
Ok.